pkg/webhook/nodenetworkconfigurationpolicy/validation.go

```diff
@@ -0,0 +1,42 @@
+package nodenetworkconfigurationpolicy
+
+import (
+	"fmt"
+	"reflect"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	shared "github.com/nmstate/kubernetes-nmstate/api/shared"
+	nmstatev1beta1 "github.com/nmstate/kubernetes-nmstate/api/v1beta1"
+)
+
+func onPolicySpecChange(policy nmstatev1beta1.NodeNetworkConfigurationPolicy, currentPolicy nmstatev1beta1.NodeNetworkConfigurationPolicy) bool {
+	return !reflect.DeepEqual(policy.Spec, currentPolicy.Spec)
+}
+
+func validateSequentialPolicyNotInProgressHook(policy nmstatev1beta1.NodeNetworkConfigurationPolicy, currentPolicy nmstatev1beta1.NodeNetworkConfigurationPolicy) []metav1.StatusCause {
+	causes := []metav1.StatusCause{}
+	currentPolicyAvailableCondition := currentPolicy.Status.Conditions.Find(shared.NodeNetworkConfigurationPolicyConditionAvailable)
+
+	if currentPolicyAvailableCondition == nil || currentPolicyAvailableCondition.Reason == shared.NodeNetworkConfigurationPolicyConditionConfigurationProgressing {
```

Comment:
I'm not sure I understand this logic. 
does `currentPolicyAvailableCondition == nil` mean "currentPolicyAvailableCondition is false"?
if so, shouldn't the logic be set with `&&` instead of `||`?

Reply:
IIUC:
1 When a policy is created, is starts with empty conditions, so in that moment, `currentPolicy.Status.Conditions.Find(shared.NodeNetworkConfigurationPolicyConditionAvailable)` returns nil
2. After a policy is created, there is a [mutating webhook](https://github.com/nmstate/kubernetes-nmstate/blob/5392270c64726dbbe8a49d9a23a8314d2a6d53c1/pkg/webhook/nodenetworkconfigurationpolicy/conditions.go#L20), that resets the status
3. Eventually the conditions are set from enactments

Reply:
So the policy can be considered in progress if either:
- it's conditions are nil
- it's conditions have empty Reason
- it's conditions Reason is `shared.NodeNetworkConfigurationPolicyConditionConfigurationProgressing`

Reply:
ack got it. thanks! 
would not insist on it, but perhaps you could name this in a informative way? like `isConditionInitializedNotSet := (currentPolicyAvailableCondition == nil)`